openfisca_core/tracers.py

```diff
@@ -242,13 +242,13 @@ def __init__(self, full_tracer):
         self._full_tracer = full_tracer
 
     def generate_graph(self, dir_path):
-        f = open(os.path.join(dir_path, 'performance.json'), 'w')
-        f.write(json.dumps(self._json()))
-        f.close()
+        with open(os.path.join(dir_path, 'performance.json'), 'w') as f:
+            f.write(json.dumps(self._json()))
+            f.close()
```

Comment:
With `with`, the file is automatically closed in the end, so that line is not necessary 🙂